File: janken.py
import random

def main():
    choices = ["グー", "チョキ", "パー"]
    print("じゃんけんゲームへようこそ！")
    print("0: グー, 1: チョキ, 2: パー")

    try:
        player = int(input("あなたの手を選んでください: "))
        if player not in [0, 1, 2]:
            print("0, 1, 2のどれかを入力してください。")
            return

        computer = random.randint(0, 2)
        print(f"あなた: {choices[player]}")
        print(f"コンピュータ: {choices[computer]}")

        if player == computer:
            print("あいこです！")
        elif (player == 0 and computer == 1) or (player == 1 and computer == 2) or (player == 2 and computer == 0):
            print("あなたの勝ちです！")
        else:
            print("あなたの負けです。")
        
        with open("debug.log", "a") as f:
            f.write(f"Player chose: {player}, Computer chose: {computer}\n")

    except ValueError:
        print("数値を入力してください。")

if __name__ == "__main__":
    main()

File: Janken.java
import java.util.Random;
import java.util.Scanner;

public class Janken {
    public static void main(String[] args) {
        String[] choices = {"グー", "チョキ", "パー"};
        Scanner scanner = new Scanner(System.in);
        Random random = new Random();

        System.out.println("じゃんけんゲームへようこそ！");
        System.out.println("0: グー, 1: チョキ, 2: パー");
        System.out.print("あなたの手を選んでください: ");

        if (scanner.hasNextInt()) {
            int player = scanner.nextInt();
            if (player < 0 || player > 2) {
                System.out.println("0, 1, 2のどれかを入力してください。");
                return;
            }

            int computer = random.nextInt(3);
            System.out.println("あなた: " + choices[player]);
            System.out.println("コンピュータ: " + choices[computer]);

            if (player == computer) {
                System.out.println("あいこです！");
            } else if ((player == 0 && computer == 1) ||
                       (player == 1 && computer == 2) ||
                       (player == 2 && computer == 0)) {
                System.out.println("あなたの勝ちです！");
            } else {
                System.out.println("あなたの負けです。");
            }
        } else {
            System.out.println("数値を入力してください。");
        }

        scanner.close();
    }
}
